hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -553,17 +546,13 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
               rowDataset -> {
                 Tuple2<RDD<GenericRecord>, RDD<String>> safeCreateRDDs = HoodieSparkUtils.safeCreateRDD(rowDataset,
                     HOODIE_RECORD_STRUCT_NAME, HOODIE_RECORD_NAMESPACE, reconcileSchema,
-                    Option.of(this.userProvidedSchemaProvider.getTargetSchema()));
+                    Option.of(schemaProvider.getTargetSchema()));
```

Comment:
we have two variables. 
schemaProvider and 
userProvidedSchemaProvider

Did you change it intentionally ?

Reply:
try to optimize ahead, there is no benefit. changed it back